logstash-core/lib/logstash/util/dead_letter_queue_factory.rb

```diff
@@ -0,0 +1,32 @@
+require 'logstash/environment'
+
+module LogStash; module Util
+  class DeadLetterQueueWriter
+
+    def initialize(inner_writer)
+      @inner_writer = inner_writer
+    end
+
+    def write(logstash_event, plugin_type, plugin_id, reason)
```

Comment:
I wonder about this API. It means the author of a plugin could make an error in referring to the plugin_type or plugin_id. Can we change this method signature to `write(logstash_event, reason)` and have the plugin_type and plugin_id be inferred, curried, or otherwise automatic without needing the caller to specify it?